src/styles/main.scss

```
@@ -1,3 +1,109 @@
-body {
+$first-day-position: 6; // 0 - is first column, 6 - the last one
+$gap: 1px;
+$days: 31;
+$day-width: 100px;
+$column: 7;
+$mounth-width: ($day-width * $column) + ($column * $gap - $gap);
+
+* {
   margin: 0;
+  padding: 0;
+}
+
+.month {
+  position: relative;
+  width: $mounth-width;
+  padding: 10px;
+  margin: 100px auto 0;
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  gap: $gap;
+
+  &__day {
+    box-sizing: border-box;
+    width: $day-width;
+    height: 100px;
+    border: 1px solid black;
+    background-color: #eee;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+  }
+
+  &__day:hover {
+    cursor: pointer;
+    background-color: pink;
+    transform: translateY(-20px);
+    transition-duration: 0.5s;
+  }
+
+  &__day:first-child {
+    margin-left: $day-width * $first-day-position + $first-day-position * $gap;
+  }
+
+  @for $index from 1 through $days {
+    .day-#{$index}::before {
```

```suggestion
    &__day:nth-child(#{$i})::before {
```
you can use `nth-child`, instead of adding extra class